swarm/storage/mru/signedupdate.go

```diff
@@ -0,0 +1,201 @@
+// Copyright 2018 The go-ethereum Authors
```

Comment:
This is not new code. This comes from splitting `update.go`.
Unfortunately, the move+changes were squashed in a rebase.
The changes include refactoring `SignedResourceUpdate` to use serializers of other structures, cleaning up serialization. As a result, you'll see `parseUpdateChunk` and `newUpdateChunk` simplified.

The reconstructed diff is here: http://www.mergely.com/RRJpC2Wz/
Sorry for the inconvenience.